### DISCUSSION TITLE
New to MicroPython and struggling to get started

### DISCUSSION THREAD

**Discussion by garryp4** [micropython/micropython#14315] [STM32 / Pyboard]
For about 15 years I have been building circuits based om Microchip 8 bit Pic's. I write the code, compile, and flash a hex file. I was expecting to learn a new language and do the same. I bought a PyBoard 1.1 and a Nucleo-L432KC. After flashing the PyBoard it no longer becomes a drive on Windows 10 but I can still make the REPL work. The Nucleo-L432K was flashed with the newest firmware from https://micropython.org/download/NUCLEO_L432KC/ and STM32CubeProgrammer and REPL works on it and it shows up as a drive on Windows. 

How can I make a program that is saved on either board? I have been reading online but can not find an answer. What am I missing?

Thanks

**Comment by scruss** [comment]
Having an MSC drive appear is not recommended. They're not true USB storage, and any attempts to write to them from the host while the dev board's CPU is busy may cause undefined behaviour.

There maybe be more Windows-centric tools, but I use [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) from the command line and an external editor. Alternatively, there's [Thonny, Python IDE for beginners](https://thonny.org/) — which isn't just for beginners — that handles board communication directly.
